### DISCUSSION TITLE
Can't Find an example for NoiseReductionEffect.

### DISCUSSION THREAD

**Discussion by GaziAkyuz** [igece/SoxSharp#25] [Q&A] [answered]
Hi!

I can't Find an example for NoiseReductionEffect. And its construction requests profile (which contains file path). However sox.Process requests file path at the sametime. 

Could you share sample please?

```
using (Sox sox = new Sox("sox.exe"))
{
  sox.Output.Type = FileType.MP3;
  sox.Effects.Add(new VolumeEffect(10, GainType.Db));
  sox.Effects.Add(new NoiseReductionEffect(new NoiseReductionEffect("inputfilepath",0.5f));

  sox.Process("inputfilepath", "test.mp3");
}
```

Not working :(

**Comment by igece** [comment, marked as answer]
Previously to NoiseReductionEffect you should add a NoiseProfileEffect. Noise Profile effect will create a file that is the file you should refer to in the NoiseReductionEffect constructor.

**Reply by eminemirhansener** [reply]
Hi
I couldn't get that what is referred to noise profile effect. It creates the noise profile but what should its file format be and it creates the profile according to what? Thanks for any reply :)
